colosseum/constants.py

```diff
@@ -342,33 +366,81 @@ def __str__(self):
 ######################################################################
 # 16.1 Indentation
 ######################################################################
+
 # text_indent
+TEXT_INDENT_CHOICES = Choices(validators=[is_length, is_percentage], explicit_defaulting_constants=[INHERIT])
 
 ######################################################################
 # 16.2 Alignment
 ######################################################################
+
 # text_align
+LEFT = 'left'
+RIGHT = 'right'
+CENTER = 'center'
+JUSTIFY = 'justify'
+
+TEXT_ALIGN_CHOICES = Choices(LEFT, RIGHT, CENTER, JUSTIFY, explicit_defaulting_constants=[INHERIT])
+
+
+def text_align_initial_value(context):
+    """Return the initial alignment valu based on direction."""
+    direction = getattr(context, 'direction')
+    if direction is LTR:
+        return LEFT
+
+    if direction is RTL:
+        return RIGHT
+
+    raise ValueError('Undefined value "{value}" for direction property!'.format(value=direction))
+
 
 ######################################################################
 # 16.3 Decoration
 ######################################################################
+
 # text_decoration
+UNDERLINE = 'underline'
+OVERLINE = 'overline'
+LINE_TRHOUGH = 'line-through'
+BLINK = 'blink'
+
+TEXT_DECORATION_CHOICES = Choices(None, UNDERLINE, OVERLINE, LINE_TRHOUGH, BLINK,
+                                  explicit_defaulting_constants=[INHERIT])
 
 ######################################################################
 # 16.4 Letter and word spacing
 ######################################################################
+
 # letter_spacing
+LETTER_SPACING_CHOICES = Choices(NORMAL, validators=[is_length], explicit_defaulting_constants=[INHERIT])
+
 # word_spacing
+WORD_SPACING_CHOICES = Choices(NORMAL, validators=[is_length], explicit_defaulting_constants=[INHERIT])
 
 ######################################################################
 # 16.5 Capitalization
 ######################################################################
+
 # text_transform
+CAPITALIZE = 'capitalize'
+UPPERCASE = 'uppercase'
+LOWERCASE = 'lowercase'
+
+TEXT_TRANSFORM_CHOICES = Choices(CAPITALIZE, UPPERCASE, LOWERCASE, None, explicit_defaulting_constants=[INHERIT])
```

Comment:
Thinking out loud, but I'm wondering about the relationship between `INHERIT` and the callable property feature you've added. They seem potentially really closely related.

Reply:
Will take notes!

Reply:
Will take notes!

Reply:
Will take notes!

Reply:
INHERIT could be a wrapper of a `str` that defines a value method as well, only that it would be used on the engine and not the CSS declaration?

Reply:
Hrm... maybe we don't need to worry about it at the level of declaration; a value of INHERIT is something we can evaluate when we perform layout - and is, itself, an actual legal constant; as opposed to text direction, which is something where you can't actually set a constant that is interpreted as "whatever text direction says" (other than using inherited value)

Reply:
Correct!